### ISSUE TITLE
Equivalent unicode identifier/variable names not treated as equivalent

### ISSUE THREAD

**Issue by kwshi** [microsoft/pyright#3030]
**Describe the bug**
Python 3 allows Unicode characters in identifier/variable names.  According to the [Python 3 documentation](https://docs.python.org/3/reference/lexical_analysis.html#identifiers), Unicode identifiers are parsed and compared under "NFKC normal form":

> All identifiers are converted into the normal form NFKC while parsing; comparison of identifiers is based on NFKC.

Unicode normal forms are documented in the Unicode standard [here].  In essence, comparisons under NFKC equivalence mean that sometimes different Unicode glyphs are considered "the same", for compatibility/accessibility reasons.  For instance, different Unicode-math-font variants of the same letter are all considered the same under NFKC:

- R (regular plain ASCII, `\U00000052`)
- 𝑅 (math italic, `\U0001d445`)
- 𝑹 (math bold italic, `\U0001d479`)
- ℝ (math blackboard bold, `\U0000211d`)
- 𝚁 (math monospace, `\U0001d681`)

(and many more).  That means that, in Python, you can write something like
```python
ℝ = "hello"
print(𝕽)
```
and have it evaluate, because both identifiers are parsed and treated as simply `R`.  However, it seems that pyright currently treats them differently and so complains that the second `𝕽` is undefined, which is inconsistent with Python's own evaluation behavior.

**To Reproduce**
See example above.

**Expected behavior**
Consistency with Python's own evaluation behavior.  (Note I'm not making any arguments about whether I like this behavior; it simply is what it is, and design questions aside it makes sense to match whatever Python does, given that it's a Python type-checker after all, right?)

**Screenshots or Code**
```python
ℝ = "hello"
print(𝕽)
```

**VS Code extension or command-line**
CLI pyright tool, neovim lsp plugin

**Comment by erictraut** [comment]
Thanks, I didn't know about this. I'll look into the possibility of adding this. It's relatively low priority though.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.222, which I just published. It will also be included in the next release of pylance.
